internal/cmd/server/common.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2024 Red Hat Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+compliance with the License. You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under the License is
+distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+implied. See the License for the specific language governing permissions and limitations under the
+License.
+*/
+
+package server
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"os"
+	"strings"
+)
+
+func processBackendToken(
+	ctx context.Context,
+	backendToken string, backendTokenFile string,
+	logger *slog.Logger) (string, error) {
+
+	// Check that the backend token and token file haven't been simultaneously provided:
+	if backendToken != "" && backendTokenFile != "" {
+		logger.ErrorContext(
+			ctx,
+			"Backend token and token file have both been provided, but they are incompatible",
+			slog.Any(
+				"flags",
+				[]string{
+					backendTokenFlagName,
+					backendTokenFileFlagName,
+				},
+			),
+			slog.String("!token", backendToken),
+			slog.String("token_file", backendTokenFile),
+		)
+		return "", errors.New("backendToken and backendTokenFile both provided")
+	}
+
+	// Read the backend token file if needed:
+	if backendToken == "" && backendTokenFile != "" {
+		backendTokenData, err := os.ReadFile(backendTokenFile)
+		if err != nil {
+			logger.ErrorContext(
+				ctx,
+				"Failed to read backend token file",
+				slog.String("file", backendTokenFile),
+				slog.String("error", err.Error()),
+			)
+			return "", errors.New("failed to read backend token file")
+		}
+		backendToken = strings.TrimSpace(string(backendTokenData))
+		logger.InfoContext(
+			ctx,
+			"Loaded backend token from file",
+			slog.String("file", backendTokenFile),
+			slog.String("!token", backendToken),
+		)
+	}
+
+	// Check that we have a token:
+	if backendToken == "" {
+		logger.ErrorContext(ctx, "Backend token or token file must be provided")
+		return "", errors.New("no token provided")
```

Comment:
Try to explain include in the error messages the names of the flags that should be used.